lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -275,16 +307,48 @@ def _on_update_status(self, event):
 logger = logging.getLogger(__name__)
 
 
+###########################################################################################
+# These are the basic types of errors:
+# * OpenSearchPluginError: base class for all plugin errors in the plugins code
+# * OpenSearchPluginConfigError: base class for all plugin errors related to config options
 class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
+    def __init__(self, msg, only_log=False):
+        super().__init__(msg)
+        self._only_log = only_log
```

Comment:
Could you explain this? what is the alternate course of action when a plugin errors?

Reply:
There are some issues that I do not want to raise, like for example if repository-gcs is not completely configured, whereas repository-s3 is. In these plugins, I raise an exception with _only_log=True.